### PR TITLE
Ensuring unit consistency in Emission generators

### PR DESCRIPTION
**Pull request by WillJRoper**
While ensuring the dust emission being generated was actually right I stumbled across some inconsistencies. To avoid these inconsistencies units are now used in the generation of spectra. Also:

- Doc strings have been cleaned up.
- `planck` has been "fixed" to produce spectral luminosity density rather than spectral radiance density. In reality this never mattered because the output was always normalised but meant the units were wrong (and would give an incorrect answer if called separately).
- Normalisation is now done consistently using `Sed` methods rather than `scipy`.

I've also updated the ruff workflow version to use the latest hash and fixed an issue introduced in #727 

## Issue Type

- Bug
- Enhancement

## Checklist
- [x] I have read the [CONTRIBUTING.md]() -->
- [x] I have added docstrings to all methods
- [x] I have added sufficient comments to all lines
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no pep8 errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by WillJRoper** [comment]
And the untis pay off: the `Greybody` had a rogue extra nu in it. The emissivity can be a function of `nu` but there is not `nu*emissivity`.

File: src/synthesizer/emission_models/dust/emission.py [hunk 1]
```diff
@@ -358,40 +377,44 @@ def _lnu(self, nu: unyt_array) -> unyt_array:
                 The unnormalised spectral luminosity density.
 
         """
+        # Ensure we have units on nu
+        if not has_units(nu):
+            raise exceptions.InconsistentArguments(
+                f"Frequency must be a given with units. (type(nu)={type(nu)})"
+            )
 
-        return nu**self.emissivity * planck(nu, self.temperature)
```
Comment (aswinpvijayan) [inline src/synthesizer/emission_models/dust/emission.py:362]: This is actually correct. That is the functional form. The units are supposedly corrected at normalisation. One thing we could do make it (nu/nu_0)**emissivity to make it unitless?

Comment (aswinpvijayan) [reply src/synthesizer/emission_models/dust/emission.py:362]: See for the different parameterisation definitions of the IR emission in: https://ui.adsabs.harvard.edu/abs/2012MNRAS.425.3094C/abstract

Comment (WillJRoper) [reply src/synthesizer/emission_models/dust/emission.py:362]: Sure, we could do that. The pure nu multiplication isn't dimensionally consistent so we should avoid it. Seems a weird parametrisation though. Can't look at the paper right now but I'll check it out.

Comment (aswinpvijayan) [reply src/synthesizer/emission_models/dust/emission.py:362]: It is defined as a proportionality in their definition, so probably they did not care about the units.

Comment (WillJRoper) [reply src/synthesizer/emission_models/dust/emission.py:362]: That'd do it 😂. We can just arbitrarily divide by Hz to maintain the previous behaviour?

**Review by stephenmwilkins — APPROVED** [review]

**Pull request merged**